code/modules/clothing/modular/modular.dm

```diff
@@ -0,0 +1,217 @@
+/**
+	Modular armor
+*/
+/obj/item/clothing/suit/modular
+	name = "Jaeger XM-02 combat exoskeleton"
+	desc = "Designed to mount a variety of modular armor components and support systems. It comes installed with light-plating and a shoulder lamp. Mount armor pieces to it by clicking on the frame with the components"
+	icon = 'icons/mob/modular/modular_armor.dmi'
+	icon_state = "underarmor_icon"
+	item_state = "underarmor"
+	flags_armor_protection = CHEST|GROIN|ARMS|LEGS
+	/// What is allowed to be equipped in suit storage
+	allowed = list(
+		/obj/item/weapon/gun/,
+		/obj/item/tank/emergency_oxygen,
+		/obj/item/storage/bible,
+		/obj/item/storage/belt/sparepouch,
+		/obj/item/storage/large_holster/machete,
+		/obj/item/weapon/claymore,
+		/obj/item/storage/belt/gun
+	)
+	flags_equip_slot = ITEM_SLOT_OCLOTHING
+	w_class = WEIGHT_CLASS_BULKY
+
+	armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 10, "bomb" = 10, "bio" = 10, "rad" = 10, "fire" = 10, "acid" = 10)
+	siemens_coefficient = 0.9
+	permeability_coefficient = 1
+	gas_transfer_coefficient = 1
+
+	actions_types = list(/datum/action/item_action/toggle)
+
+	/// An assoc list of stat mods
+	var/list/stat_mod
+
+	/// The human that is wearing the armor
+	var/mob/living/carbon/human/owner
+
+	/// Attachment slots for chest armor
+	var/obj/item/armor_module/armor/slot_chest
+	/// Attachment slots for arm armor
+	var/obj/item/armor_module/armor/slot_arms
+	/// Attachment slots for leg armor
+	var/obj/item/armor_module/armor/slot_legs
+
+	/** Installed modules */
+	/// How many modules you can have
+	var/max_modules = 1
+	/// What modules are installed
+	var/list/obj/item/armor_module/attachable/installed_modules
+	/// What storage is installed
+	var/obj/item/armor_module/storage/installed_storage
+	/// Holder for the actual storage implementation
+	var/obj/item/storage/internal/storage
+
+	///List of allowed types [/obj/item/armor_module] that can be installed into this modular armor.
+	var/allowed_armor_path = list(
+		/obj/item/armor_module,
+	)
+ 	///List of allowed types [/obj/item/armor_module] that can be installed into this modular armor.
+	var/allowed_modules_path = list(
+		/obj/item/armor_module,
+	)
+	///List of allowed types [/obj/item/armor_module] that can be installed into this modular armor.
+	var/allowed_storage_path = list(
+		/obj/item/armor_module,
+	)
+
+	/// Misc stats
+	light_strength = 4
+
+
+/obj/item/clothing/suit/modular/Initialize()
+	. = ..()
+	installed_modules = list()
+
+
+/obj/item/clothing/suit/modular/Destroy()
+	QDEL_NULL(slot_chest)
+	QDEL_NULL(slot_arms)
+	QDEL_NULL(slot_legs)
+
+	QDEL_NULL(installed_modules)
+	QDEL_NULL(installed_storage)
+	QDEL_NULL(storage)
+	return ..()
```

Comment:
Make sure to null `owner`, though it's currently an unused.
`installed_modules` and the other lists cannot be qdeleted. Just null them instead.
Or do a `for` loop to qdel and remove each entry from the list before nulling the list.